### DISCUSSION TITLE
Divergent Score Reporting for Solver run

### DISCUSSION THREAD

**Discussion by terbrack** [TimefoldAI/timefold-solver#836] [General]
Hi,
I am facing the following issue regarding the score reporting of a solving run:
there is a mismatch between the solver score printed after solving ended and the constraint list with associated scores, see:

> 2024-05-07 18:03:14,008 [main] INFO  Solving started: time spent (44), best score (0hard/-73572medium/-73226soft), environment mode (REPRODUCIBLE), move thread count (NONE), random (JDK with seed 0).

> 2024-05-07 18:03:14,016 [main] INFO  Construction Heuristic phase (0) ended: time spent (53), best score (0hard/-73572medium/-73226soft), score calculation speed (0/sec), step total (0).

> 2024-05-07 18:05:15,505 [main] INFO  Local Search phase (1) ended: time spent (121542), best score (0hard/-72401medium/-71917soft), score calculation speed (288/sec), step total (13).

> 2024-05-07 18:05:15,509 [main] INFO  Solving ended: time spent (121543), best score (0hard/-72401medium/-71917soft), score calculation speed (288/sec), phase total (2), environment mode (REPRODUCIBLE), move thread count (NONE).

>  constraint matchCount                    score
>  1 constraint_one          3     0hard/0medium/-3soft
>  2    constraint_two         2     0hard/0medium/-2soft
>  3          constraint_three         16  0hard/-2811medium/0soft
>  4      constraint_four         90 0hard/-71831medium/0soft
>  5            constraint_five         90 0hard/0medium/-71912soft
> Explanation of score (0hard/-74642medium/-71917soft)

What makes me wonder is the mismatch between the reported score at the end of local search phase, (0hard/-72401medium/-71917soft) , and the summed up score in the constraint match summary, (0hard/-74642medium/-71917soft).

Do you have an idea why this could happen?
Thanks,
Hajo

**Comment by triceo** [comment]
Hey there, my first bet is a score corruption. Run for a long time with `FULL_ASSERT` environment mode and if it fails, you have a bug in your constraints and/or variable listeners and/or custom moves.
